### ISSUE TITLE
OpenVPN generates invalid configuration for net30 [non-legacy]

### ISSUE THREAD

**Issue by joni1993** [opnsense/core#7228]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**
When transitioning from legacy to the new configuration interface, the generated config is invalid for net30 configurations and won't start

**To Reproduce**

1. Go to 'Generate OpenVPN Server config in new UI'
2. Use 'net30' as 'Topology'
3. Apply

**Expected behavior**

OpenVPN instance starts normally 

**Relevant log files**

```
/usr/local/opnsense/scripts/openvpn/ovpn_service_control.php: The command '/usr/local/sbin/openvpn --config '/var/etc/openvpn/instance-84b64fa6-83fe-41e9-9862-9d7816065e4b.conf'' returned exit code '1', the output was ''
Options error: --client-connect requires --mode server
```

**Environment**

OPNsense 24.1.1-amd64
FreeBSD 13.2-RELEASE-p9
OpenSSL 3.0.13

**Comment by AdSchellevis** [comment]
https://github.com/opnsense/core/commit/66fd0e4699ae06bc9a1038ecdb441a6c6b71a875

**Comment by joni1993** [comment]
Thank you - I did not find this.

But one problem remains: the generated ifconfig-pool section causes "IPv4 pool size is too small (1), must be at least 2" on startup. Removing that config line at least brings up the instance (the config line is also not generated by the legacy interface)

**Comment by joni1993** [comment]
Also setting verify-client-certificate to none results in `Options error: --verify-client-cert none|optional must be used with --management-client-auth, an --auth-user-pass-verify script, or plugin`

**Comment by AdSchellevis** [comment]
I'm not sure we can fix that, it depends on the exact settings being configured (which aren't in the ticket).

**Comment by joni1993** [comment]
Yeah the second issue is more of like a validation issue probably.

The config is the most basic s2s one:
```
Role: Server
Protocol: UDP
Port: 1194
Type: tun
Server: x.x.x.8/30
Topology: net30
Certificate: <selected>
Certificate Depth: Do not check
TLS Statis key: [auth] <selected>
Auth: SHA256
Data Cipher: AES-256-GCM
Data Cipher-Fallback: AES-256-GCM
Local Network: Y.Y.Y.Y/24
Remote Network: Z.Z.Z.Z/16
```

**Comment by AdSchellevis** [comment]
a bit as expected, the server /30 is not allowed, when it's about a single (peer to peer) tunnel, you could try p2p mode as in that case the /30 does seem to work.  I've added https://github.com/opnsense/core/commit/66fd0e4699ae06bc9a1038ecdb441a6c6b71a875 to have some option for /30 peer to peers, we should probably exclude net30 in these cases as this only works for p2p (net30 expects room for at least 2 remote clients it seems).

Available options can be found at https://openvpn.net/community-resources/reference-manual-for-openvpn-2-6/

I'll restrict the validation and close this ticket, nothing we can do about this (without running into other issues)

**Comment by joni1993** [comment]
Another thing i found is the fallback cipher being multi value - according to OpenVPN it's single value only and the instance refuses to start when multiples are selected.

**Comment by AdSchellevis** [comment]
@joni1993 seems I missed that one `data-ciphers-fallback` suggest multiple, documentation suggest single indeed. I'll change this as well, hardly used anyway.

**Comment by AdSchellevis** [comment]
@joni1993 ok, that should be this one https://github.com/opnsense/core/commit/0a17929ec77be0d4865c32b621ecf5893ae18128

**Comment by joni1993** [comment]
Great! Thanks for your super fast fixes! - That brings my S2S up :)

I will migrate Road Warrior instance too and maybe get back to the issue tracker and maybe put some PR up - know i see where most of the bits and pieces are :)

**Comment by AdSchellevis** [comment]
@joni1993 you're welcome, some of these validations are easily missed but do improve usability quite a lot. Just open a new ticket if you run into something else.

**Comment by joni1993** [comment]
Just need to figure out why with the new config i don't get traffic through the tunnel:
```
MULTI: bad source address from client [X.X.X.X], packet dropped
```

With the legacy config my routed setup worked just fine...

**Comment by AdSchellevis** [comment]
not using the address received via the pool perhaps? in tls mode the server needs to know the client in order to send packets to it.

**Comment by joni1993** [comment]
Hmm.. the remote client itself uses the correct address. But none of the non-natted traffic that is/should/was pushed through the tunnel is "accepted".

I noticed in the routes I only have the remote client IP (which is probably expected for P2P) - but none of the remote networks specified in the server.

I will probably have to revert to the legacy config for now and dig into openvpn docs...

**Comment by joni1993** [comment]
Hmm.. the remote client itself uses the correct address. But none of the non-natted traffic that is/should/was pushed through the tunnel is "accepted".

I noticed in the routes I only have the remote client IP (which is probably expected for P2P) - but none of the remote networks specified in the server.

I will probably have to revert to the legacy config for now and dig into openvpn docs...

**Comment by laozhoubuluo** [comment]
> Just need to figure out why with the new config i don't get traffic through the tunnel:
> 
> ```
> MULTI: bad source address from client [X.X.X.X], packet dropped
> ```
> 
> With the legacy config my routed setup worked just fine...

We recently encountered this problem. After careful analysis, we found that the problem originated from the OpenVPN configuration file generated by OPNsense.
After careful comparison with the legacy mode, we found that the original OpenVPN used the official p2p mode (mode p2p), and the tunnel can go through any IP address, which is suitable for scenarios where NAT is not allowed and dynamic routing protocols are used for networking. More infomations can be found at https://openvpn.net/community-resources/reference-manual-for-openvpn-2-6/ .
In the new version, the use of server mode (mode server) is hard-coded in the scenario of configuring the /30 network segment. The server mode cannot support this scenario because it involves multiple clients, resulting in this problem.
